File: Programming101/week0/count_vowels.py
from count_substrings import count_substrings


def count_vowels(str_):
    vowels = ['a', 'e', 'i', 'o', 'u', 'y', 'A', 'E', 'I', 'O', 'U', 'Y']
    result = 0
    for ch in vowels:
        result += count_substrings(str_, ch)
    return result


def main():
    print(count_vowels("Python"))
    print(count_vowels("Theistareykjarbunga"))
    print(count_vowels("grrrrgh!"))
    print(count_vowels("Github is the second best thing that happend to programmers, after the keyboard!"))
    print(count_vowels("A nice day to code!"))


if __name__ == '__main__':
    main()

File: FrontendJavaScript/week0/2-A-Set-Of-Problems/set1/countVowels.js
"use strict";

var countVowels = function(str) {
  var vowels = "aeiouy".split("");
  var times = 0;
  str = str.toLowerCase();
  str = str.split("");
  str.forEach(function(letter) {
    if (vowels.indexOf(letter) != -1) {
      times += 1;
    }
  });
  return times;
};
exports.countVowels = countVowels;

// console.log(countVowels("JavaScript") === 3);
// console.log(countVowels("Theistareykjarbunga") === 8);
// console.log(countVowels("grrrrgh!") === 0);
// console.log(countVowels("Github is the second best thing that happend to programmers, after the keyboard!") === 22);
// console.log(countVowels("A nice day to code!") === 8);
